dot/rpc/websocket.go

```diff
@@ -191,12 +192,18 @@ func (h *HTTPServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func (h *HTTPServer) registerSubscription(conn *websocket.Conn, reqID float64, subscriptionType int) (uint32, error) {
+func (h *HTTPServer) registerSubscription(conn *websocket.Conn, reqID float64, subscriptionType int, params interface{}) (uint32, error) {
 	wssub := h.serverConfig.WSSubscriptions
 	sub := uint32(len(wssub)) + 1
+	pA := params.([]interface{})
+	filter := make(map[string]bool)
+	for _, param := range pA {
+		filter[param.(string)] = true
+	}
 	wss := &WebSocketSubscription{
 		WSConnection:     conn,
 		SubscriptionType: subscriptionType,
+		Filter:           filter,
 	}
 	wssub[sub] = wss
 	h.serverConfig.WSSubscriptions = wssub
```

Comment:
the WSSubscriptions should be stored in the HTTPServer struct (or in a separate wsServer struct) not in the config